core/domain/topic_validators.py

```diff
@@ -0,0 +1,585 @@
+
+        return base_model_validators.VALIDATION_MODE_NON_STRICT
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        snapshot_model_ids = [
+            '%s-%d' % (item.id, version) for version in python_utils.RANGE(
+                1, item.version + 1)]
+        skill_ids = item.uncategorized_skill_ids
+        for subtopic in item.subtopics:
+            skill_ids = skill_ids + subtopic['skill_ids']
+        skill_ids = list(set(skill_ids))
+        canonical_story_ids = [
+            reference['story_id']
+            for reference in item.canonical_story_references]
+        additional_story_ids = [
+            reference['story_id']
+            for reference in item.additional_story_references]
+        return [
+            base_model_validators.ExternalModelFetcherDetails(
+                'topic_commit_log_entry_ids',
+                topic_models.TopicCommitLogEntryModel,
+                ['topic-%s-%s'
+                 % (item.id, version) for version in python_utils.RANGE(
+                     1, item.version + 1)]),
+            base_model_validators.ExternalModelFetcherDetails(
+                'topic_summary_ids', topic_models.TopicSummaryModel, [item.id]),
+            base_model_validators.ExternalModelFetcherDetails(
+                'topic_rights_ids', topic_models.TopicRightsModel, [item.id]),
+            base_model_validators.ExternalModelFetcherDetails(
+                'snapshot_metadata_ids',
+                topic_models.TopicSnapshotMetadataModel, snapshot_model_ids),
+            base_model_validators.ExternalModelFetcherDetails(
+                'snapshot_content_ids', topic_models.TopicSnapshotContentModel,
+                snapshot_model_ids),
+            base_model_validators.ExternalModelFetcherDetails(
+                'story_ids', story_models.StoryModel,
+                canonical_story_ids + additional_story_ids),
+            base_model_validators.ExternalModelFetcherDetails(
+                'skill_ids', skill_models.SkillModel, skill_ids),
+            base_model_validators.ExternalModelFetcherDetails(
+                'subtopic_page_ids',
+                subtopic_models.SubtopicPageModel,
+                ['%s-%s' % (
+                    item.id, subtopic['id']) for subtopic in item.subtopics])]
+
+    @classmethod
+    def _validate_canonical_name_is_unique(cls, item):
+        """Validate that canonical name of the model unique.
+
+        Args:
+            item: datastore_services.Model. TopicModel to validate.
+        """
+        topic_models_list = topic_models.TopicModel.query().filter(
+            topic_models.TopicModel.canonical_name == (
+                item.canonical_name)).filter(
+                    topic_models.TopicModel.deleted == False).fetch() # pylint: disable=singleton-comparison
+        topic_model_ids = [
+            topic_model.id
+            for topic_model in topic_models_list if topic_model.id != item.id]
```

Comment:
Why is this check required? We are expecting topic IDs to be unique, so can't we raise an error if length of `topic_models_list` is greater than 1? If multiple models have same IDs, that would also be an issue, right?

Reply:
Hi @aks681, This looks like an [existing code](https://github.com/oppia/oppia/pull/10967/files#diff-7c0a61daaa262e962cd70544379ab4b330963a02e4ddd4929a02ab3c37e831e6L2948), we aren't expecting this PR to optimize the code. Can we work on this issue separately?

Note: We expect this PR to only move code from one file to another new file. (and the contributor doesn't have enough knowledge of the codebase to work on these issues.)

Reply:
@aks681 I believe changes to functionality, as @DubeySandeep said, may be outside the scope of this refactor. I can create a separate issue to document this optimization.

Reply:
Ah ok, didn't know that. Yup, can you create an issue for this then?

Reply:
@aks681, We have already created an issue at #11085, PTAL! (I've also left a comment for you in the issue thread)